packages/purpose-process/src/services/purposeService.ts

```diff
@@ -1340,75 +1339,51 @@ export function purposeServiceBuilder(
 
 export type PurposeService = ReturnType<typeof purposeServiceBuilder>;
 
-const authorizeRiskAnalysisForm = ({
-  purpose,
-  producerId,
-  organizationId,
-  tenantKind,
-  activeProducerDelegation,
-}: {
-  purpose: Purpose;
-  producerId: TenantId;
-  organizationId: TenantId;
-  tenantKind: TenantKind;
-  activeProducerDelegation: Delegation | undefined;
-}): { purpose: Purpose; isRiskAnalysisValid: boolean } => {
-  if (
-    organizationId === purpose.consumerId ||
-    organizationId === producerId ||
-    organizationId === activeProducerDelegation?.delegateId
-  ) {
-    if (purposeIsDraft(purpose)) {
-      const isRiskAnalysisValid = isRiskAnalysisFormValid(
-        purpose.riskAnalysisForm,
-        false,
-        tenantKind
-      );
-      return { purpose, isRiskAnalysisValid };
-    } else {
-      return { purpose, isRiskAnalysisValid: true };
-    }
-  } else {
-    return {
-      purpose: { ...purpose, riskAnalysisForm: undefined },
-      isRiskAnalysisValid: false,
-    };
-  }
-};
-
 const getOrganizationRole = async ({
-  eserviceId,
-  organizationId,
+  eservice,
   producerId,
   consumerId,
+  authData,
   readModelService,
 }: {
-  eserviceId: EServiceId;
-  organizationId: TenantId;
+  eservice: EService;
   producerId: TenantId;
   consumerId: TenantId;
+  authData: AuthData;
   readModelService: ReadModelService;
 }): Promise<Ownership> => {
-  if (producerId === consumerId && organizationId === producerId) {
+  if (producerId === consumerId && authData.organizationId === producerId) {
     return ownership.SELF_CONSUMER;
-  } else if (producerId !== consumerId && organizationId === consumerId) {
-    return ownership.CONSUMER;
   }
 
-  const activeProducerDelegation = await readModelService.getActiveDelegation(
-    eserviceId,
-    delegationKind.delegatedProducer
-  );
-
-  if (
-    (activeProducerDelegation &&
-      organizationId === activeProducerDelegation.delegateId) ||
-    (!activeProducerDelegation && organizationId === producerId)
-  ) {
+  try {
+    assertRequesterCanActAsProducer(
+      eservice,
+      authData,
+      await readModelService.getActiveProducerDelegationByEserviceId(
+        eservice.id
+      )
+    );
     return ownership.PRODUCER;
+  } catch {
+    /* empty */
   }
 
-  throw organizationNotAllowed(organizationId);
+  try {
+    assertRequesterCanActAsConsumer(
+      { eserviceId: eservice.id, consumerId },
+      authData,
+      await readModelService.getActiveConsumerDelegationByPurpose({
+        eserviceId: eservice.id,
+        consumerId,
+      })
+    );
+    return ownership.CONSUMER;
+  } catch {
+    /* empty */
+  }
+
+  throw organizationNotAllowed(authData.organizationId);
```

Comment:
Maybe it's clearer to do this as we do in `assertRequesterIsAllowedToRetrieveRiskAnalysisDocument` with a nested try-catch:

```ts
try {
   // Can act as producer...
} catch {
    // Can act as consumer...
}
```
no need for a last empty catch and throw since these assertions already throw the `organizationNotAllowed` error